securedrop_client/api_jobs/downloads.py

```diff
@@ -284,11 +289,21 @@ def call_decrypt(self, filepath: str, session: Session = None) -> str:
         '''
         with NamedTemporaryFile('w+') as plaintext_file:
             self.gpg.decrypt_submission_or_reply(filepath, plaintext_file.name, is_doc=False)
-            set_message_or_reply_content(
-                model_type=Reply,
-                uuid=self.uuid,
-                session=session,
-                content=plaintext_file.read())
+            try:
+                set_message_or_reply_content(
+                    model_type=Reply,
+                    uuid=self.uuid,
+                    session=session,
+                    content=plaintext_file.read())
+            finally:
```

Comment:
you can remove this cleanup since we use a temporary file that automatically gets removed (see how when you remove this block of code, there is no decryption file?

Reply:
The temporary file is gone, but because I'm doing the work in subdirectories under data_dir, instead of at the top level, the folder named after the UUID of the message or reply will be left behind, empty. This cleanup keeps those from accumulating and possibly affecting performance.

Reply:
The temporary file is gone, but because I'm doing the work in subdirectories under data_dir, instead of at the top level, the folder named after the UUID of the message or reply will be left behind, empty. This cleanup keeps those from accumulating and possibly affecting performance.

Reply:
ah i see this makes sense. 

just out of curiosity, why did you chose not to decrypt to the same uuid directory, e.g. instead of:
```
Message/baf388ff-dfbd-45e1-a629-bf9e761b1ac8
Reply/baf388ff-dfbd-45e1-a629-bf9e761b1ac8
File/baf388ff-dfbd-45e1-a629-bf9e761b1ac8/example_submission.pdf
``` 

why not:
```
baf388ff-dfbd-45e1-a629-bf9e761b1ac8/example_submission.pdf
baf388ff-dfbd-45e1-a629-bf9e761b1ac8/<temporary message and reply files that are removed>
```

Reply:
ah maybe it's to avoid having an empty uuid directory if there are only messages and replies?

but i guess you could just decrypt them to the data dir and then there wouldn't need to be any folder creation and deletion cleanup since they will be NamedTemporaryFIles.

Reply:
The reason for including the class in the path is that UUIDs are only unique within a table, so if we were writing them into the root of the data directory, we could theoretically see collisions. I know, I know, _UUIDs_, but this approach uses the database constraints to guarantee it can't happen.

Reply:
so it makes sense to use the uuid folder for files because the files might have the same filename, but I don't think you need to create the uuid folder and subfolders for replies and messages since they are `NamedTemporaryFile`s (they won't collide).

Reply:
i guess i wasn't considering earlier how perhaps having the UUID folder might make it easier to programatically find messages and replies that failed to decrypt (rather than having to look at the filename and use journalist designation). i think that's a good enough reason to keep the new folder structure you've introduced. i do think we should all discuss this further during standup, because we have [one directory structure in the export vm](https://docs.google.com/document/d/1qoQMMGjB9d9mHelZg_lncgiMjjB1BxIvZot_2Y6KjqM), aka devices vm, and another one here now and it might make sense to standardize. 

did you test what happens when the same source (same uuid) submits the same file twice? one nice thing about storing the downloaded file in a folder named after the original server filename is that we can keep a separate copy for each submission. let's chat more in standup!

Reply:
here is the export archive format explained in the readme: https://github.com/freedomofpress/securedrop-export/blob/master/README.md#export-archive-format

Reply:
I have not tested submitting the same file twice as the same source, but each submission would have its own UUID, and that's what's used in the directory path, not the source's UUID. So one source can send the same filename multiple times, multiple sources can send the same filename, and they won't all be written to the same file in `$SDC_HOME/data`.

Reply:
To recap our discussion from standup this morning, there were generally three ideas kicking around (we went with the third idea):

1. keep storing downloaded messages and replies in the data directory and if they fail to decrypt, that's where we'll find them, and store submission files in the data directory until they are successfully decrypted into a directory with a unique name, either UUID or the server filename.

Pros: 

* If messages and replies successfully decrypt, we don't have to delete any folders

Cons: 

* files are not organized under a journalist designation, so files that are part of the same conversation would have to be collected in various places if we want to one day export multiple files and messages.

* maintaining two different data directory structures in securedrop-export and securedrop-client, more cognitive overload

2. store all messages and replies in their respective `<Type>/<UUID>` directories

Pros: 

* the UUID never changes, so we don't have to change the name of the directory when the journalist designation changes

Cons: 

(same cons as above, also a note about how we already have to change the filename if the journalist designation changes)

3. maintain the same data directory structure as securedrop-export (see https://github.com/freedomofpress/securedrop-export/blob/master/README.md#export-archive-format):

```
├── cytotoxic payer
│   ├── 1-cytotoxic-payer-msg
│   │   └── file-to-export-1.txt
│   ├── 2-cytotoxic-payer-msg
│   │   └── file-to-export-2.txt
│   └── 3-cytotoxic-payer-doc
│   │   └── original-filename.doc
├── grandiloquent pasteboard
│   └── 1-grandiloquent-pasteboard-doc
│   │   └── original-filename.pdf
└── snug seek
```

Pros: 

* same directory structure as we (plan to) use in the securedrop-export, less to remember, and if we find a problem with the structure we can easily fix it in both places

* files are organized under a journalist designation, so files that are part of the same conversation would be easier to collect for a multi-file export or a full-source conversation export

Cons:

* We have to change the name of the folders in addition to the files if the journalist designation changes

---

One additonal thought I'm having now is that we _might_ want to consider storing messages directly under the journalist designation name, instead of create a subdirectory of the server filename, since they will always be text files and have the same server filename as the subdirecotry. So instead of:

```
```
├── cytotoxic payer
│   ├── 1-cytotoxic-payer-msg
│   │   └── file-to-export-1.txt
│   ├── 2-cytotoxic-payer-msg
│   │   └── file-to-export-2.txt
│   └── 3-cytotoxic-payer-doc
│   │   └── original-filename.doc
├── grandiloquent pasteboard
│   └── 1-grandiloquent-pasteboard-doc
│   │   └── original-filename.pdf
└── snug seek
```

I'm proposing:
```
├── cytotoxic payer
│   ├── 1-cytotoxic-payer-msg
│   ├── 2-cytotoxic-payer-msg
│   └── 3-cytotoxic-payer-doc
│   │   └── original-filename.doc
├── grandiloquent pasteboard
│   └── 1-grandiloquent-pasteboard-doc
│   │   └── original-filename.pdf
└── snug seek
```

But this is a minor detail, and shouldn't hold up this PR since this has yet to be implemented in securdrop-export.

Ok, that's the recap, plus one additional thought!

Reply:
One of the requirements in #714 was to remove `File.original_filename`, instead replacing `File.filename` with the submission's original filename when the submission was unpacked. That's going to make the hierarchy above impossible, since we won't have `1-grandiloquent-pasteboard-doc` in the `File` once it's been processed.

Keeping the `original_filename` column would enable this structure.

Reply:
ok so we discussed this a bit and @creviera is gonna pick up https://github.com/freedomofpress/securedrop/issues/4304 to remove the renaming of `journalist_designation` entirely, we'll pick into the next core/server minor release. One thing we thought about when we discussed this is that the metadata sync (and file rename) and file/message/reply download can occur concurrently, so it's definitely worth removing the rename to avoid racing behavior (i.e. to avoid guarding against such behavior). This way we can just use the securedrop-export structure everywhere. 

so: for the remainder of this review, we can assume that this will be tested with a version of securedrop server that lacks journalist_designation renaming

Reply:
and for those not part of the discussion today, we agreed that it makes sense to stick with idea #3 (see recap of all the three ideas we discussed and their pros and cons from Wednesday's standup: https://github.com/freedomofpress/securedrop-client/pull/737#discussion_r372573628)

Reply:
(ignore the `#3` link... forgot that would link to an issue)